python/pyarrow/feather.py

```diff
@@ -151,7 +151,18 @@ def write_feather(df, dest, compression=None, compression_level=None,
             df = df.to_dense()
 
     if _pandas_api.is_data_frame(df):
-        table = Table.from_pandas(df, preserve_index=False)
+        """
+        Feather v1 creates a new column in the resultant Table to
+        store index information if index type is not RangeIndex
+        """
+        if version == 1 and type(df.index) is not _pandas_api.pd.RangeIndex:
+            preserve_index = False
```

Comment:
```suggestion
        if version == 1:
            preserve_index = False
```

I would maybe keep the current behaviour for `preserve_index=False` for all cases when the user passes `version=1`. Since a user would only use this for legacy reasons anyway, it seems better to not change the behaviour here.